src/main/java/io/micronaut/gradle/graalvm/MicronautGraalPlugin.java

```diff
@@ -135,6 +137,14 @@ public void apply(Project project) {
                     nativeImageTestTask.setDescription("Runs tests against a native image build of the server. Requires the server to allow the port to configurable with 'micronaut.server.port'.");
             })));
 
+            TaskProvider<GenerateResourceConfigFile> generateResourceConfig = configureResourcesFileGeneration(project, tasks);
+            tasks.withType(NativeImageTask.class).configureEach(nativeImage -> {
+                // This isn't great. Ideally the configuration file directories should be an input
+                // of the native image task directly, not something we patch
+                DirectoryProperty outputDirectory = generateResourceConfig.get().getOutputDirectory();
+                nativeImage.getInputs().files(generateResourceConfig);
+                nativeImage.args("-H:ConfigurationFileDirectories="+outputDirectory.get().getAsFile().getAbsolutePath());
```

Comment:
This way of setting the arguments in the native image task doesn't work and the directory is not included in the native image. I've added `--verbose` to my test application and I can't see the directory included.

I've tried as a workaround defining manually the arg in my app and it works. Also I can see the directory being included when generated the native image.

```
nativeImage {
    imageName("basic-app")
    args = [
        "--verbose",
        "-H:ConfigurationFileDirectories=/home/ivan/workspaces/micronaut-graal-tests/micronaut-basic-app/build/generated/resources/graalvm"
    ]
}
```

Reply:
@melix can we include it in the `META-INF/native-image/resource-config.json` inside the JAR directly? This way the JAR has the config and arguments are not needed

Reply:
I see, that's why I don't like how the current `NativeImage` task is setup. It basically erases whatever is configured, which isn't good, it should be more explicit. Do you have any guidance what I could use to test the actual generated image?

Reply:
When you run native image with `--verbose` it spits out the computed arguments from all of the inputs (config files, explicit arguments etc.) I guess that is the only way today that I know of.

Reply:
@graemerocher that should probably work, yes!

Reply:
@ilopmar I have pushed another change which should fix this and added a test, which passed locally: https://scans.gradle.com/s/zojwohjssznwu/tests#:-test-io.micronaut.gradle-micronautgraalpluginspec-generate-graalvm-resource-config.json-with-openapi-and-resources-included-0

I didn't use the `jar` approach because we should avoid it if possible, but if you see other problems I'm happy to reconsider.